File: ledgex_backend/src/main/java/com/ledgex/auth/dto/AuthResponse.java
package com.ledgex.auth.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AuthResponse {

    private String token;
    private String tokenType;
    private Long userId;
    private String email;
    private String firstName;
    private String lastName;
}

File: ledgex_frontend/src/utils/authResponse.js
/**
 * Extracts token and user from a successful auth API response.
 * Backend wraps payloads in ApiResponse: { success, message, data: AuthResponse }.
 */
export function parseAuthResponse(response) {
  const authData = response.data?.data ?? response.data

  if (!authData?.token) {
    throw new Error('Invalid auth response: missing token')
  }

  const { token, email, firstName, lastName, userId } = authData

  const user = {
    userId,
    email,
    firstName,
    lastName,
    name: [firstName, lastName].filter(Boolean).join(' ') || email,
  }

  return { token, user }
}
